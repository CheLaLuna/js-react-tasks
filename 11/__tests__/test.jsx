import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';

import MyForm from '../src/MyForm.jsx';

test('MyForm', async () => {
  const { asFragment } = render(<MyForm />);

  expect(screen.getByRole('textbox', { name: /email/i })).toBeInTheDocument();
  expect(screen.getByLabelText(/password/i)).toBeInTheDocument();
  expect(screen.getByRole('textbox', { name: /address/i })).toBeInTheDocument();
  expect(screen.getByRole('textbox', { name: /city/i })).toBeInTheDocument();
  expect(screen.getByRole('combobox', { name: /country/i })).toBeInTheDocument();
  expect(screen.getByRole('checkbox', { name: /accept rules/i })).toBeInTheDocument();
  expect(screen.getByRole('button', { name: /sign in/i })).toBeInTheDocument();

  await userEvent.click(screen.getByRole('button', { name: /sign in/i }));

  expect(asFragment()).toMatchSnapshot();
  expect(screen.getByRole('button', { name: /back/i })).toBeInTheDocument();

  await userEvent.click(screen.getByRole('button', { name: /back/i }));
  expect(screen.getByRole('form')).toHaveFormValues({
    acceptRules: false,
    address: '',
    city: '',
    country: '',
    email: '',
    password: '',
  });

  await userEvent.click(screen.getByRole('checkbox', { name: /accept rules/i }));
  await userEvent.type(screen.getByRole('textbox', { name: /city/i }), 'My City');
  await userEvent.type(screen.getByRole('textbox', { name: /email/i }), 'test@email.com');
  await userEvent.type(screen.getByLabelText(/password/i), 'mysuperpass');
  await userEvent.selectOptions(screen.getByRole('combobox', { name: /country/i }), ['russia']);
  await userEvent.click(screen.getByRole('button', { name: /sign in/i }));

  expect(asFragment()).toMatchSnapshot();

  await userEvent.click(screen.getByRole('button', { name: /back/i }));

  expect(screen.getByRole('form')).toHaveFormValues({
    acceptRules: true,
    address: '',
    city: 'My City',
    country: 'russia',
    email: 'test@email.com',
    password: 'mysuperpass',
  });
});
