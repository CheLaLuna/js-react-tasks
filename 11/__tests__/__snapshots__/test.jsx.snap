// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyForm 1`] = `
<DocumentFragment>
  <div>
    <button
      class="btn btn-primary"
      type="button"
    >
      Back
    </button>
    <table
      class="table"
    >
      <tbody>
        <tr>
          <td>
            acceptRules
          </td>
          <td>
            false
          </td>
        </tr>
        <tr>
          <td>
            address
          </td>
          <td />
        </tr>
        <tr>
          <td>
            city
          </td>
          <td />
        </tr>
        <tr>
          <td>
            country
          </td>
          <td />
        </tr>
        <tr>
          <td>
            email
          </td>
          <td />
        </tr>
        <tr>
          <td>
            password
          </td>
          <td />
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`MyForm 2`] = `
<DocumentFragment>
  <div>
    <button
      class="btn btn-primary"
      type="button"
    >
      Back
    </button>
    <table
      class="table"
    >
      <tbody>
        <tr>
          <td>
            acceptRules
          </td>
          <td>
            true
          </td>
        </tr>
        <tr>
          <td>
            address
          </td>
          <td />
        </tr>
        <tr>
          <td>
            city
          </td>
          <td>
            My City
          </td>
        </tr>
        <tr>
          <td>
            country
          </td>
          <td>
            russia
          </td>
        </tr>
        <tr>
          <td>
            email
          </td>
          <td>
            test@email.com
          </td>
        </tr>
        <tr>
          <td>
            password
          </td>
          <td>
            mysuperpass
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
